prow/plugins/config_test.go

```diff
@@ -1437,3 +1439,84 @@ func TestConfigUpdaterResolve(t *testing.T) {
 		})
 	}
 }
+
+func TestEnabledReposForPlugin(t *testing.T) {
+	pluginsYaml := []byte(`
+orgA:
+ excluded_repos:
+ - repoB
+ plugins:
+ - pluginCommon
+ - pluginNotForRepoB
+orgA/repoB:
+ plugins:
+ - pluginCommon
+ - pluginOnlyForRepoB
+`)
+	var p Plugins
+	err := yaml.Unmarshal(pluginsYaml, &p)
+	if err != nil {
+		t.Errorf("cannot unmarshal plugins config: %v", err)
+	}
+	cfg := Configuration{
+		Plugins: p,
+	}
+	testCases := []struct {
+		name              string
+		wantOrgs          []string
+		wantRepos         []string
+		wantExcludedRepos map[string]sets.String
+	}{
+		{
+			name:              "pluginCommon",
+			wantOrgs:          []string{"orgA"},
+			wantRepos:         []string{"orgA/repoB"},
+			wantExcludedRepos: map[string]sets.String{"orgA": {}},
+		},
+		{
+			name:              "pluginNotForRepoB",
+			wantOrgs:          []string{"orgA"},
+			wantRepos:         nil,
+			wantExcludedRepos: map[string]sets.String{"orgA": {"orgA/repoB": {}}},
+		},
+		{
+			name:              "pluginOnlyForRepoB",
+			wantOrgs:          nil,
+			wantRepos:         []string{"orgA/repoB"},
+			wantExcludedRepos: map[string]sets.String{},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			orgs, repos, excludedRepos := cfg.EnabledReposForPlugin(tc.name)
+			if diff := cmp.Diff(tc.wantOrgs, orgs); diff != "" {
+				t.Errorf("expected wantOrgs differ from actual: %s", diff)
+			}
+			if diff := cmp.Diff(tc.wantRepos, repos); diff != "" {
+				t.Errorf("expected repos differ from actual: %s", diff)
+			}
+			if diff := cmp.Diff(tc.wantExcludedRepos, excludedRepos); diff != "" {
+				t.Errorf("expected excludedRepos differ from actual: %s", diff)
+			}
+		})
+	}
+}
+
+func TestPluginsUnmarshalFailed(t *testing.T) {
+	badPluginsYaml := []byte(`
+orgA:
+ excluded_repos = [ repoB ]
+ plugins:
+ - pluginCommon
+ - pluginNotForRepoB
+orgA/repoB:
+ plugins:
+ - pluginCommon
+ - pluginOnlyForRepoB
+`)
+	var p Plugins
+	err := yaml.Unmarshal(badPluginsYaml, &p)
```

Comment:
Sorry I must haven't got enough coffee for the morning, didn't notice this test didn't use `UnmarshalJSON` method. Please modify the test to use it

Reply:
Sorry, fixed!